### PR TITLE
CPM-633: Enable identifier configuration for family attribute requirements

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**

This PR aims to enable the configuration of sku for family attribute requirements.
It can now be enabled or disabled for completeness

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Component/Updater/FamilyUpdater.php [hunk 1]
```diff
@@ -295,9 +295,7 @@ protected function createAttributeRequirementsByChannel(
                     $attributeCode
                 );
             }
-            if (AttributeTypes::IDENTIFIER !== $attribute->getType()) {
```
Comment (cemorin) [inline src/Akeneo/Pim/Structure/Component/Updater/FamilyUpdater.php:298]: Before, when importing a family without sku as attribute_requirement, sku was automatically added. Isn't it a too big of a breaking change?

File: src/Akeneo/Pim/Structure/Bundle/Query/InternalApi/Family/DefaultSqlGetRequiredAttributesMasks.php [hunk 2]
```diff
@@ -66,6 +66,13 @@ public function fromFamilyCodes(array $familyCodes): array
     AND family.code IN (:familyCodes)
     AND attribute.attribute_type IN (:attributeTypes)
 GROUP BY family.code, channel_code, locale_code
+UNION
+SELECT family.code as family_code, channel_code, locale_code, JSON_ARRAY()
```
Comment (pierallard) [inline src/Akeneo/Pim/Structure/Bundle/Query/InternalApi/Family/DefaultSqlGetRequiredAttributesMasks.php:70]: Can you add AS mask at the end of the line to better understand what is this JSON_ARRAY ?

**Review by pierallard — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Component/Validator/Constraints/FamilyRequirementsValidator.php [hunk 3]
```diff
@@ -100,26 +76,4 @@ protected function validateRequiredAttributes(FamilyInterface $family, FamilyReq
             }
         }
     }
-
-    /**
-     * @param FamilyInterface $family
-     *
-     * @return string[]
-     */
-    protected function getMissingChannelCodes(FamilyInterface $family)
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Component/Validator/Constraints/FamilyRequirementsValidator.php:109]: I guess we don't need the channelRepository dependency anymore, you can remove it from the constructor and the DI

File: tests/back/Pim/Structure/Specification/Component/Updater/FamilyUpdaterSpec.php [hunk 4]
```diff
@@ -232,19 +231,6 @@ function it_updates_a_family_by_removing_an_attribute(
...
-    ) {
-        $values = [
-            'code' => 'mycode',
-        ];
-
-        $family->setCode('mycode')->shouldBeCalled();
-        $family->setAttributeRequirements(Argument::any())->shouldNotBeCalled();
-
-        $this->update($family, $values, []);
-    }
```
Comment (mmetayer) [inline tests/back/Pim/Structure/Specification/Component/Updater/FamilyUpdaterSpec.php:246]: this test is still relevant, why did you delete it?

**Review by mmetayer — APPROVED** [review]

File: tests/back/Pim/Structure/Specification/Component/Updater/FamilyUpdaterSpec.php (hunk 4 above)
Comment (cemorin) [reply tests/back/Pim/Structure/Specification/Component/Updater/FamilyUpdaterSpec.php:246]: True, it will be in the next PR :+1:
